connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/internal/columnNodeSupport.scala

```diff
@@ -212,9 +221,68 @@ object ColumnNodeToProtoConverter extends (ColumnNode => proto.Expression) {
   }
 }
 
+object ColumnNodeUtil {
+  def collectReferences(nodes: Seq[ColumnNode]): Seq[proto.Relation] = {
```

Comment:
Shall we add some API to ColumnNode that basically traverses the tree for you? For example node.collect()/node.foreach()... The reason I'd prefer that over this is that it should protect us a bit against new ColumnNode types that we don't cover in this method.